pkg/adaptor/cloud/libvirt/libvirt.go

```diff
@@ -116,55 +128,196 @@ func uploadIso(isoFile string, isoVolName string, libvirtClient *libvirtClient)
 
 }
 
-func CreateDomain(ctx context.Context, libvirtClient *libvirtClient, v *vmConfig) (result *createDomainOutput, err error) {
+func getGuestForArchType(caps *libvirtxml.Caps, arch string, virttype string) (*libvirtxml.CapsGuest, error) {
```

Comment:
@CarstenLeue let me ask you two small changes here: 1) rename `virttype` with `ostype` to be "complaint" with the output of `virsh capabilities` 2) have a constant for "hvm" (that you can pass to ` getCanonicalMachineName()` later.

Ah, if you could document the new functions... I know that the libvirt's implementation has poor documentation but let's try to improve it;